src/wallet/walletdb.h

```diff
@@ -287,11 +290,47 @@ class WalletBatch
     //! Abort current transaction
     bool TxnAbort();
 private:
-    BerkeleyBatch m_batch;
     WalletDatabase& m_database;
 };
 
 //! Compacts BDB state so that wallet.dat is self-contained (if there are changes)
 void MaybeCompactWalletDB();
 
+//! State of scanning the wallet during ReadKeyValue
+class CWalletScanState {
+public:
+    unsigned int nKeys{0};
+    unsigned int nCKeys{0};
+    unsigned int nWatchKeys{0};
+    unsigned int nKeyMeta{0};
+    unsigned int m_unknown_records{0};
+    bool fIsEncrypted{false};
+    bool fAnyUnordered{false};
+    std::vector<uint256> vWalletUpgrade;
+    std::map<OutputType, uint256> m_active_external_spks;
+    std::map<OutputType, uint256> m_active_internal_spks;
+    std::map<uint256, DescriptorCache> m_descriptor_caches;
+    std::map<std::pair<uint256, CKeyID>, CKey> m_descriptor_keys;
+    std::map<std::pair<uint256, CKeyID>, std::pair<CPubKey, std::vector<unsigned char>>> m_descriptor_crypt_keys;
+
+    CWalletScanState() {
+    }
+};
+
+//! Unserialize a give Key-Value pair and load it into the wallet and wallet scan state
+bool ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
+             CWalletScanState &wss, std::string& strType, std::string& strErr) EXCLUSIVE_LOCKS_REQUIRED(pwallet->cs_wallet);
```

Comment:
When building with clang --enable-debug -Werror I'm seeing the following compiler error:

```cpp
./wallet/walletdb.h:322:112: error: member access into incomplete type 'CWallet'
             CWalletScanState &wss, std::string& strType, std::string& strErr) EXCLUSIVE_LOCKS_REQUIRED(pwallet->cs_wallet);
                                                                                                               ^
./wallet/ismine.h:14:7: note: forward declaration of 'CWallet'
class CWallet;
      ^
1 error generated.
```

Reply:
(It may not be directly related to this PR; saw a similar build error in another wallet PR today.)